src/client/updateClientMetaInfo.js

```diff
@@ -1,64 +1,75 @@
-import updateTitle from './updaters/updateTitle'
-import updateTagAttributes from './updaters/updateTagAttributes'
-import updateTags from './updaters/updateTags'
+import { updateAttribute, updateTag, updateTitle } from './updaters'
 
-export default function _updateClientMetaInfo (options = {}) {
+const getTag = (tags, tag) => {
+  if (!tags[tag]) {
+    tags[tag] = document.getElementsByTagName(tag)[0]
+  }
+
+  return tags[tag]
+}
+
+/**
+ * Performs client-side updates when new meta info is received
+ *
+ * @param  {Object} newInfo - the meta info to update to
+ */
+export default function updateClientMetaInfo(options = {}, newInfo) {
   const { ssrAttribute } = options
 
-  /**
-   * Performs client-side updates when new meta info is received
-   *
-   * @param  {Object} newInfo - the meta info to update to
-   */
-  return function updateClientMetaInfo (newInfo) {
-    const htmlTag = document.getElementsByTagName('html')[0]
-    // if this is not a server render, then update
-    if (htmlTag.getAttribute(ssrAttribute) === null) {
-      // initialize tracked changes
-      const addedTags = {}
-      const removedTags = {}
+  // only cache tags for current update
+  const tags = {}
+
+  const htmlTag = getTag(tags, 'html')
+
+  // if this is not a server render, then update
+  if (htmlTag.getAttribute(ssrAttribute) === null) {
+    // initialize tracked changes
+    const addedTags = {}
+    const removedTags = {}
+
+    Object.keys(newInfo).forEach((type) => {
+      // ignore these
+      if ([
```

Comment:
Making this array constant would increase code readability and runtime perf by not creating a new object each time.

Reply:
Regarding runtime perf, you are probably gonna hate me for this but please check these tests: https://jsperf.com/const-vs-inline-array

Although there is a difference its almost negligible (~1%). Especially if you look at the absolute performance of the operation, 830 mio / sec is quite fast (on my machine with Chrome 69 its 830 mio/s for inline and 875 mio/s for const).

For php I know for sure that it actually doesnt matter what you do as the php engine is smart enough to recognize such variables as constants and will treat them as constants for you. It seems v8 is also that kind of smart.